compiler/src/iree/compiler/Codegen/Common/FoldTensorSubsetIntoVectorTransferOps.cpp

```diff
@@ -204,11 +205,99 @@ class FoldInsertSliceIntoTransferWrite final
   }
 };
 
+/// Fold tensor.extract_slice into vector.transfer_write if
+///   1. The vector.transfer_write op has only one use.
+///   2. All the offests of the tensor.extract_slice op are zeros.
+///   3. The vector.transfer_write op does not have masks.
+///   4. The vector.transfer_write op writes to a tensor.empty op.
+///
+/// E.g.:
+///
+/// ```
+/// %0 = vector.transfer_write %v, %t[%a, %b, %c]
+///   {in_bounds = [true, true, true]}
+///   : vector<1x64x128xf16>, tensor<1x64x128xf16>
+/// %extracted_slice = tensor.extract_slice %0[0, 0, 0] [1, %3, 128] [1, 1, 1]
+///   : tensor<1x64x128xf16> to tensor<1x?x128xf16>
+/// ```
+/// is rewritten to:
+/// ```
+/// %1 = vector.transfer_write %v, %t2[%a, %b, %c]
+///   {in_bounds = [true, false, true]}
+///   : vector<4x5xf32>, tensor<?x?xf32>
+/// ```
```

Comment:
Looking at dims in this example - are these correct? For example, `vector<1x64x128xf16>` vs `vector<4x5xf32>`?